src/core/socket_handler.rs

```diff
@@ -17,7 +17,7 @@ use anyhow::Result;
 
 #[derive(Debug)]
 pub struct SocketHandler {
-    socket_name: String,
+    pub socket_name: String,
```

Comment:
Any reason why we're making this public?

Reply:
In process we need to know the socket name so we can then pass it into the process so that it knows which socket to connect to.
https://github.com/polysoft1/polychat-ipc/pull/11/files#diff-17a09042d453007a26a13e912c25f7863231fb7aeb2f8fe2a9bb803ef2819754R28

I think it makes sense for SocketHandler to be the single source of truth for the socket name.
The only alternative I see is to make a getter that just clones socket_name. What do you think?

Reply:
`pub` allows this to be changed if `SocketHandler` is mutable somewhere, we should make a getter that returns a reference to the socket name, that way we avoid creating an entirely new string and there's no issues with mutability. 

So `get_socket_name(&self) -> &String` would be the signature, which cannot be updated

Reply:
Switched to getter.